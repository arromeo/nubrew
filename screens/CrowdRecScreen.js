// make sure this gets deleted at the end and figure out how to set-up proxy
const port = require('../dev_port.json');

import React from 'react';
import { ScrollView, StyleSheet, View, Text, ActivityIndicator } from 'react-native';
import BeerSearch from './search/BeerSearch.js';
import GoToCamera from './components/GoToCamera.js';

export default class CrowdRecommendation extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      favorites: null,
      loading: true,
    }
  }

  componentDidMount() {

    let url = `${port.DEV_PORT}/api/crowdrecommendations`;
    fetch(url)
      .then(res => res.json())
      .then(data => this.setState ({
        favorites: data.result,
        loading: false,
      }))
      .catch(error => {
        console.error(error);
      });
  }

  render() {
    const { navigate } = this.props.navigation;

    return (
      <View style={{flex: 1}}>
        <ScrollView style={styles.container}>
          {this.state.loading &&
            <View style={styles.spinner}>
              <ActivityIndicator size={100} color="orange" />
            </View> 
          }
          {!this.state.loading && 
            <View>
              <BeerSearch data={this.state.favorites} navigate={navigate}/>
              <View style={{height: 100}}/>
            </View>
          }
         </ScrollView>
         {!this.state.loading &&
          <GoToCamera navigate={navigate} />
          }
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 15,
    backgroundColor: '#fff',
  },
  spinner: {
    margin: 200,
    alignSelf: 'center'
  }
});
