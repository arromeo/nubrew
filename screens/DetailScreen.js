// make sure this gets deleted at the end and figure out how to set-up proxy
const port = require('../dev_port.json');

import React from 'react';
import { ScrollView, View, StyleSheet, Text } from 'react-native';
import BeerSearch from './search/BeerSearch.js';
import StoreSearch from './search/StoreSearch.js';
import BrewerySearch from './search/BrewerySearch.js';
import EventSearch from './search/EventSearch.js';
import SearchComponent from './search/SearchComponent.js';

export default class DetailScreen extends React.Component {

  render() {
    const searchDatabase = (value, category) => {
    const data = {
      category: category,
      keywords: value,
    }
    return fetch(`${port.DEV_PORT}/api/details`, 
      {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify(data),
      })
      .then(res => res.json())
      .then(data => {
        if (data.searchCategory === "None") {
          this.setState({
            searchResultCategory: "None",
          })
        } else {
          this.setState({
            searchResult: data.searchResult,
            searchResultCategory: data.searchResultCategory,
            loading: false,
          })
        }})
      .catch((error) => {
        console.error(error);
      })
    }
  
    return (
      <ScrollView style={styles.container}>
        <SearchComponent input={this.state.input} changeInput={changeInput} pickerValue={this.state.pickerValue} pickCategory={pickCategory}/>
          
        <Button 
          title="Search"
          onPress={() => searchDatabase(this.state.input, this.state.pickerValue)}
          />

        {!this.state.loading &&
          <View style={styles.searchContainer}>
            {this.state.searchResultCategory === "Beer" &&
              <BeerSearch data={this.state.searchResult} styles={styles}/>
            }
            {this.state.searchResultCategory === "Brewery" &&
              <BrewerySearch data={this.state.searchResult} styles={styles}/>
            }
            {this.state.searchResultCategory === "Store" &&
              <StoreSearch data={this.state.searchResult} styles={styles}/>
            }
            {this.state.searchResultCategory === "Event" &&
              <EventSearch data={this.state.searchResult} styles={styles}/>
            }
            {this.state.searchResultCategory === "None" &&
              <Text>No results could be found.</Text>
            }
          </View>
        }
      </ScrollView>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },

})