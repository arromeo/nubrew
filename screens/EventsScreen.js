// make sure this gets deleted at the end and figure out how to set-up proxy
const port = require('../dev_port.json');

import React from 'react';
import { ScrollView, StyleSheet, View, Text, FlatList, TouchableOpacity, Image, ActivityIndicator } from 'react-native';
import EventList from './home/EventList.js';
import GoToCamera from './goto/GoToCamera.js';

export default class Events extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      events: [],
      loading: true,
    }
  }

  componentDidMount() {

    let url = `${port.DEV_PORT}/api/events`;
    fetch(url)
      .then(res => res.json())
      .then(data => {
        this.setState ({
          events: data,
          loading: false,
        })
      })
      .catch(error => {
        console.error(error);
      });
  }

  render() {
    const { navigate } = this.props.navigation;

    return (
      <View style={{flex: 1}}>
        <ScrollView style={styles.container}>
          {this.state.loading &&
            <View style={styles.spinner}>
              <ActivityIndicator size={100} color="orange" />
            </View> 
          }
          {!this.state.loading && 
            <View>
                <FlatList
                  data={this.state.events.result}
                  keyExtractor={item => item.id.toString()}
                  renderItem={({item}) => 
                <TouchableOpacity
                  style={[styles.eventContainer, styles.homeScreenFilename]}
                  onPress={() => {
                    navigate('Detail', {
                    id: item.id,
                    category: "Event"
                  });
                }}>
                <View style={[styles.eventDetailsContainer, styles.homeScreenFilename]}>
                <View>
                  <Image
                    style={{width: 75, height: 75}}
                    source={{uri: item.store_img_url }}
                  />
                </View>
                <View style={styles.eventDetails}>
                  <Text style={styles.eventTitle}>{item.event_name}</Text>
                  <Text>{item.store_name} @ {item.time}</Text>
                  <Text>{item.city}, {item.province}</Text>
                </View>
              </View>
            </TouchableOpacity>
          }/>
            </View>
          }
        </ScrollView>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 15,
    backgroundColor: '#fff',
  },
  headerFont: {
    color: 'black',
    fontWeight: 'bold',
    marginBottom: 20,
    flex: 1,
    textAlign: 'center',
    justifyContent: 'center',
  },
  contentContainer: {
    paddingBottom: 5,
    flexDirection: "row",
    justifyContent: "center",
    alignItems: 'flex-start',
  },
  eventDetailsContainer: {
    flexDirection: "row",
    flex: 1,
    margin: 5,
  },
  eventDetails: {
    marginLeft: 15,
    justifyContent: "center"
  },
  eventTitle: {
    fontWeight: "bold"
  },
  spinner: {
    margin: 200,
    alignSelf: 'center'
  }
});