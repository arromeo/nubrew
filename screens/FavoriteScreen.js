// make sure this gets deleted at the end and figure out how to set-up proxy
const port = require('../dev_port.json');

import React from 'react';
import { ScrollView, StyleSheet, View, Text, ActivityIndicator } from 'react-native';
import FavoriteList from './favorites/FavoriteList.js';
import GoToCamera from './goto/GoToCamera.js';


export default class FindScreen extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      favorites: null,
      loading: true,
    }
  }

  componentDidMount() {
    // need to figure out how to pass userid through different stacks
    let url = `${port.DEV_PORT}/api/user/${this.props.screenProps.user_id}/favorites`;
    fetch(url)
      .then(res => res.json())
      .then(data => this.setState ({
        favorites: data.result,
        loading: false,
      }))
      .catch(error => {
        console.error(error);
      });
  }

  render() {
    const { navigate } = this.props.navigation;

    return (
      <View style={{flex: 1}}>
        <ScrollView style={styles.container}>
          {this.state.loading &&
            <View style={styles.spinner}>
              <ActivityIndicator size={100} color="orange" />
            </View> 
          }
          {!this.state.loading && 
            <View>
              <FavoriteList data={this.state.favorites} navigate={navigate} crowdFavorite={false} />
            </View>
          }
          <View style={{height: 100}}/>
        </ScrollView>
        <GoToCamera navigate={navigate}/>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 15,
    backgroundColor: '#fff',
  },
  headerFont: {
    color: 'black',
    fontWeight: 'bold',
    marginBottom: 20,
    flex: 1,
    textAlign: 'center',
    justifyContent: 'center',
  },
  spinner: {
    margin: 150,
    alignSelf: 'center'
  }
});
