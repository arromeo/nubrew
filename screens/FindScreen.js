// make sure this gets deleted at the end and figure out how to set-up proxy
const port = require('../dev_port.json');

import React from 'react';
import { ScrollView, StyleSheet, View, Text, Button } from 'react-native';
import BeerSearch from './search/BeerSearch.js';
import StoreSearch from './search/StoreSearch.js';
import BrewerySearch from './search/BrewerySearch.js';
import EventSearch from './search/EventSearch.js';
import SearchComponent from './search/SearchComponent.js';

export default class FindScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      input: "",
      pickerValue: "Beer",
      searchResult: null,
      searchResultCategory: null,
      loading: true,
    }
  }

  render() {

    const changeInput = (event) => {
      this.setState({
        input: event,
      })
    }

    const pickCategory = (event) => {
      this.setState({
        pickerValue: event,
      })
    }

    const searchDatabase = (value, category) => {
      const data = {
        category: category,
        keywords: value,
      }
      return fetch(`${port.DEV_PORT}/api/find`, 
        {
          method: "POST",
          headers: { "Content-Type": "application/json" },
          body: JSON.stringify(data),
        })
      .then(res => res.json())
      .then(data => {
        if (data.searchCategory === "None") {
          this.setState({
            searchResultCategory: "None",
          })
        } else {
          this.setState({
            searchResult: data.searchResult,
            searchResultCategory: data.searchResultCategory,
            input: "",
            loading: false,
          })
        }})
      .catch((error) => {
        console.error(error);
      })
    }

    // if search category is beer... do... else if store... do... else if brewery... do...
    return (
      <ScrollView style={styles.container}>
        <SearchComponent input={this.state.input} changeInput={changeInput} pickerValue={this.state.pickerValue} pickCategory={pickCategory}/>
          
        <Button 
          title="Search"
          onPress={() => searchDatabase(this.state.input, this.state.pickerValue)}
          />

        {!this.state.loading &&
          <View style={styles.searchContainer}>
            {this.state.searchResultCategory === "Beer" &&
              <BeerSearch data={this.state.searchResult}/>
            }
            {this.state.searchResultCategory === "Brewery" &&
              <BrewerySearch data={this.state.searchResult}/>
            }
            {this.state.searchResultCategory === "Store" &&
              <StoreSearch data={this.state.searchResult}/>
            }
            {this.state.searchResultCategory === "Event" &&
              <EventSearch data={this.state.searchResult}/>
            }
            {this.state.searchResultCategory === "None" &&
              <Text>No results could be found.</Text>
            }
          </View>
        }
      </ScrollView>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 15,
    backgroundColor: '#fff',
  },
  searchResultContainer: {
    borderWidth: 1,
    borderStyle: "dotted",
    flexDirection: "column",
    width: '90%',
    margin: 10,
    justifyContent: 'center', 
    alignItems: 'center',
  },
});
