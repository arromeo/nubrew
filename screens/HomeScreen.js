// make sure this gets deleted at the end and figure out how to set-up proxy
const port = require('../dev_port.json');

import React from 'react';
import {
  Image,
  ScrollView,
  StyleSheet,
  Text,
  View,
  FlatList,
  TouchableOpacity,
} from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { WebBrowser } from 'expo';
import { MonoText } from '../components/StyledText';

export default class HomeScreen extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      event: null,
      recommendedBeer: null,
      loading: true,
    }
  }
  static navigationOptions = {
    title: 'NuBrew',
  };

  randomGen = (data) => {
    return Math.floor(Math.random() * data.length);
  }

  componentDidMount() {
    fetch(`${port.DEV_PORT}/api/index`)
      .then(res => res.json())
      .then(data => this.setState ({
        event: data.result.events,
        recommendedBeer: data.result.featured_beer[this.randomGen(data.result.featured_beer)],
        loading: false
      }))
      .catch(error => {
        console.error(error);
      });
  }

  render() {
    const event = this.state.event;
    const recommendedBeer = this.state.recommendedBeer;
    const { navigate } = this.props.navigation;

    return (
      <View style={styles.container}>
        {this.state.loading && 
          <View><Text>LoadingScreen goes here</Text></View>
        }
        {!this.state.loading && 
          <ScrollView style={styles.container}>
            <View style={styles.contentContainer}>

              <FlatList
                data={event}
                keyExtractor={item => item.id.toString()}
                renderItem={({item}) => 
                <TouchableOpacity
                  style={[styles.eventContainer, styles.homeScreenFilename]}
                  onPress={() => {
                    navigate({
                      routeName: 'Find',
                      params: {
                        type: 'Event',
                        item
                      }
                    });}}
                  >
                  <View style={[styles.eventDetailsContainer, styles.homeScreenFilename]}>
                    <Text>{item.event_name} at {item.store_name}</Text>
                    <MonoText style={styles.codeHighlightText}>{item.details.split('').slice(0, 60).join("")}...</MonoText>
                  </View>
                </TouchableOpacity>
              }/>
            </View>

            <View style={styles.contentContainer}>
              <TouchableOpacity 
                style={[styles.recommendationContainer, styles.homeScreenFilename]}
                onPress={() => {
                  navigate({
                    routeName: 'Find',
                    params: {
                      type: 'Beer',
                      recommendedBeer
                    }
                  });}}
                >
                <Image
                  source={
                    __DEV__
                      ? require('../assets/images/robot-dev.png')
                      : require('../assets/images/robot-prod.png')
                  }
                  size={100}
                />
                <Text>{recommendedBeer.brewery_name}'s</Text>
                <Text>{recommendedBeer.beer_name}</Text>
                <MonoText style={styles.codeHighlightText}>IBU: {recommendedBeer.ibu}</MonoText>
                <MonoText style={styles.codeHighlightText}>ABV: {recommendedBeer.abv}</MonoText>
              </TouchableOpacity>
      
              <TouchableOpacity 
                style={[styles.recommendationContainer, styles.homeScreenFilename]}
                onPress={() => {
                  navigate({
                    routeName: 'Find',
                    params: {
                      type: 'CrowdRecommendations',
                    }
                  });}}
              >
                <Ionicons name="md-medal" size={100} color="black"/>
                <Text>Crowd</Text>
                <Text>Recommendations</Text>
              </TouchableOpacity>
            </View>

          </ScrollView>
        }

      </View>
    );
  }

  _handleLearnMorePress = () => {
    WebBrowser.openBrowserAsync('https://docs.expo.io/versions/latest/guides/development-mode');
  };

  _handleHelpPress = () => {
    WebBrowser.openBrowserAsync(
      'https://docs.expo.io/versions/latest/guides/up-and-running.html#can-t-see-your-changes'
    );
  };
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
  contentContainer: {
    paddingTop: 15,
    paddingBottom: 15,
    borderWidth: 0.5,
    borderColor: "black",
    borderWidth: 1,
    borderStyle: "dotted",
    flexDirection: "row",
    justifyContent: 'center',
    alignItems: 'center',
    flex: 1,
  },
  eventContainer: {
    flex: 1,
    flexDirection: "row",
  },
  eventDetailsContainer: {
    borderWidth: 1,
    borderStyle: "dotted",
    flexDirection: "column",
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    margin: 10,
  },
  recommendationContainer: {
    borderWidth: 1,
    borderStyle: "dotted",
    flexDirection: "column",
    width: '45%',
    margin: 10,
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    textAlign: 'center',
  },
  bottomOptionsContainer: {
    borderWidth: 1,
    borderStyle: "dotted",
    flexDirection: "column",
    width: '28.33%',
    margin: 10,
  },
  welcomeImage: {
    width: 50,
    height: 40,
    resizeMode: 'contain',
    marginTop: 3,
    marginLeft: -10,
  },
  eventImage: {
    width: 100,
    height: 80,
    resizeMode: 'contain',
    margin: 10,
    width: '20%',
  },
  homeScreenFilename: {
    marginVertical: 7,
  },
  optionScreenFilename: {
    marginVertical: 7,
  },
  getStartedText: {
    fontSize: 17,
    color: 'rgba(96,100,109, 1)',
    lineHeight: 24,
    textAlign: 'center',
  },
});
