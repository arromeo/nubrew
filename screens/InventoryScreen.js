// make sure this gets deleted at the end and figure out how to set-up proxy
const port = require('../dev_port.json');

import React from 'react';
import { ScrollView, View, ActivityIndicator, StyleSheet } from 'react-native';
import BeerSearch from './search/BeerSearch.js';

export default class InventoryScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      data: null,
    }
  }

  componentDidMount() {
    const navigationParams = this.props.navigation.state.params;
    const data = {
      id: navigationParams.id,
      category: navigationParams.category,
    }
    fetch(`${port.DEV_PORT}/api/:location_id/inventory`, {
      method: 'POST',
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(data),
    })
      .then((res) => res.json())
      .then((data) => {
        console.log(data);
        this.setState({
          data: data.result,
        })
      })
      .catch(err => {
        console.error(err);
      })
  }

  render() {
    const { navigate } = this.props.navigation;

    return (
      <ScrollView style={styles.container}>
        {!this.state.data &&
          <View style={styles.spinner}>
            <ActivityIndicator size={100} color="orange" />
          </View> 
        }
        {this.state.data &&
          <BeerSearch data={this.state.data} navigate={navigate}/>
        }
      </ScrollView>
    )
  }
}

const styles = StyleSheet.create({
  spinner: {
    margin: 200,
    alignSelf: 'center'
  },
})