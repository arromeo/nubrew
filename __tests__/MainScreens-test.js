import 'react-native';
import React from 'react';
import HomeScreen from '../screens/HomeScreen';
import FindScreen from '../screens/FindScreen';
import RateDrinkScreen from '../screens/RateDrinkScreen';
import FavoriteScreen from '../screens/FavoriteScreen';
import ProfileScreen from '../screens/ProfileScreen';
import CrowdRecScreen from '../screens/CrowdRecScreen';
import DetailScreen from '../screens/DetailScreen';
import RecommendationsScreen from '../screens/RecommendationsScreen';
import EventsScreen from '../screens/EventsScreen';
import InventoryScreen from '../screens/InventoryScreen';
import RecommendationListScreen from '../screens/RecommendationListScreen';
import renderer from 'react-test-renderer';

const props = {
  navigation: {
    navigate: function() {
      return;
    },
    state: {
      params: {
        category: "Beer",
        id: 1,
        data: {
          meridians: {
            latitude: 49.270621,
            longitude: -123.135671
          }
        }
      }
    }
  },
  screenProps: {
    user_id: 1,
    currentSearch: "Pilsner",
    currentSearchCategory: "Beer",
    favorites: [
      {
      beer_id: 1,
      category: "Porter",
      beer_name: "Nitro Milk Stout",
      brewery_name: "Left Hand Brewing Company",
      ibu: 25,
      abv: "6.0%",
      img_url: "https://products3.imgix.drizly.com/ci-left-hand-milk-stout-nitro-1cc9bd255865b63b.png?auto=format%2Ccompress&fm=jpeg&q=20"
      },
      {
      beer_id: 3,
      category: "Hefeweizen",
      beer_name: "Weissbier",
      brewery_name: "Erdinger Brewery",
      ibu: 10,
      abv: "5.3%",
      img_url: "https://www.totalwine.com/media/sys_master/twmmedia/h7c/he3/8820602306590.png"
      }
      ]
  }

}


describe('Main screens render properly', () => {

  it('renders the HomeScreen', async () => {
    const tree = renderer.create(<HomeScreen navigation={props.navigation}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the FindScreen', async () => {
    const tree = renderer.create(<FindScreen navigation={props.navigation} screenProps={props.screenProps}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the RateDrinkScreen', async () => {
    const tree = renderer.create(<RateDrinkScreen navigation={props.navigation}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the FavoriteScreen', async () => {
    const tree = renderer.create(<FavoriteScreen navigation={props.navigation} screenProps={props.screenProps}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the ProfileScreen', async () => {
    const tree = renderer.create(<ProfileScreen navigation={props.navigation} screenProps={props.screenProps}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the CrowdRecScreen', async () => {
    const tree = renderer.create(<CrowdRecScreen navigation={props.navigation}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the DetailScreen', async () => {
    const tree = renderer.create(<DetailScreen navigation={props.navigation} screenProps={props.screenProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the RecommendationsScreen', async () => {
    const tree = renderer.create(<RecommendationsScreen navigation={props.navigation} screenProps={props.screenProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the EventsScreen', async () => {
    const tree = renderer.create(<EventsScreen navigation={props.navigation} screenProps={props.screenProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the InventoryScreen', async () => {
    const tree = renderer.create(<InventoryScreen navigation={props.navigation} screenProps={props.screenProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the RecommendationListScreen', async () => {
    const tree = renderer.create(<RecommendationListScreen navigation={props.navigation} screenProps={props.screenProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
