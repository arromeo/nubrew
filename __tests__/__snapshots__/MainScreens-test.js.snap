// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main screens render properly renders the CrowdRecScreen 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;

exports[`Main screens render properly renders the DetailScreen 1`] = `null`;

exports[`Main screens render properly renders the EventsScreen 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;

exports[`Main screens render properly renders the FavoriteScreen 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#61170E",
        "borderRadius": 50,
        "bottom": 20,
        "flex": 1,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 19,
        "width": 50,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#FFBC02",
            "fontSize": 32,
          },
          undefined,
          Object {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`Main screens render properly renders the FindScreen 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#61170E",
        "borderRadius": 50,
        "bottom": 20,
        "flex": 1,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 19,
        "width": 50,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#FFBC02",
            "fontSize": 32,
          },
          undefined,
          Object {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`Main screens render properly renders the HomeScreen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "margin": 200,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="orange"
      hidesWhenStopped={true}
      size={100}
    />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#61170E",
        "borderRadius": 50,
        "bottom": 20,
        "flex": 1,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 19,
        "width": 50,
      }
    }
  >
    <Text />
  </View>
  <Text
    style={
      Object {
        "bottom": 20,
        "fontSize": 15,
        "fontWeight": "bold",
        "left": 20,
        "position": "absolute",
      }
    }
  >
    Please drink responsibly.
  </Text>
</View>
`;

exports[`Main screens render properly renders the InventoryScreen 1`] = `null`;

exports[`Main screens render properly renders the ProfileScreen 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "margin": 150,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="orange"
      hidesWhenStopped={true}
      size={100}
    />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#61170E",
        "borderRadius": 50,
        "flex": 0.1,
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFBC02",
        }
      }
    >
      Logout
    </Text>
  </View>
</View>
`;

exports[`Main screens render properly renders the RateDrinkScreen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 15,
        "fontWeight": "bold",
        "justifyContent": "center",
        "paddingTop": 10,
        "textAlign": "center",
        "width": "90%",
      }
    }
  >
    Sorry! This feature is not available if camera permission is not provided.
  </Text>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "red",
          "fontSize": 50,
        },
        undefined,
        Object {
          "fontFamily": "ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#61170E",
        "borderRadius": 25,
        "flex": 0.2,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginTop": 10,
        "opacity": 1,
        "padding": 5,
        "width": "75%",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "yellow",
            "fontSize": 50,
          },
          undefined,
          Object {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`Main screens render properly renders the RecommendationListScreen 1`] = `null`;

exports[`Main screens render properly renders the RecommendationsScreen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "margin": 150,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="orange"
      hidesWhenStopped={true}
      size={100}
    />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#61170E",
        "borderRadius": 50,
        "bottom": 20,
        "flex": 1,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 19,
        "width": 50,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#FFBC02",
            "fontSize": 32,
          },
          undefined,
          Object {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
